test/reduce-e2e/testdata/join-on-map-pipeline.yaml

```diff
@@ -0,0 +1,58 @@
+apiVersion: numaflow.numaproj.io/v1alpha1
+kind: Pipeline
+metadata:
+  name: join-on-map
```

Comment:
Do we need a reduce vertex for join-on-map test? If no, should we remove the reduce vertex to make the test simpler and consider moving it out of the reduce test suite?

Reply:
We do need the reduce vertex for this test. Here I was explicitly trying to verify the case of the Joined Map vertex as far as the watermark, since the Reduce Vertex downstream helps to verify that the Watermark was passed successfully.